### ISSUE TITLE
Claims to Windows Token Service

### ISSUE THREAD

**Issue by Mr-Nivek** [MicrosoftDocs/OfficeDocs-SharePoint#1524]
Claims to Windows Token Service service on all MinRoles, except custom causes the health check rule 'Server role configuration isn't correct' to state the server is non-compliant.  This stops and disables the service.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5bbcd95d-56ce-63b6-804f-14098ce1953b
* Version Independent ID: 78166e52-c779-2ad6-7b1f-cc3bcde47515
* Content: [Description of MinRole and associated services in SharePoint Servers 2016 and 2019](https://docs.microsoft.com/en-us/SharePoint/administration/description-of-minrole-and-associated-services-in-sharepoint-server-2016)
* Content Source: [SharePoint/SharePointServer/administration/description-of-minrole-and-associated-services-in-sharepoint-server-2016.md](https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/blob/live/SharePoint/SharePointServer/administration/description-of-minrole-and-associated-services-in-sharepoint-server-2016.md)
* Product: **sharepoint-server-itpro**
* GitHub Login: @MikePlumleyMSFT
* Microsoft Alias: **mikeplum**

**Comment by scanum** [comment]
@Mr-Nivek Thank you for submitting feedback and contributing to the docs. We are currently investigating this.

**Comment by get-itips** [comment]
Hello @Mr-Nivek 
How are you? Hope you are doing great.
Is there any specific documentation change that you see needs to be added to this article?
Thanks

**Comment by Mr-Nivek** [comment]
Claims to Windows Token Service service on all MinRoles causes the health check rule 'Server role configuration isn't correct' to state the server is non-compliant.

Either the health check rule is broken or the documentation needs to be updated. I am leaning towards the health check rule is wrong.

**Comment by get-itips** [comment]
Hello @MikePlumleyMSFT 
Can you help me confirm if the documentation is wrong?
It seems that running Claims to Windows Token Service service on all MinRoles except custom is non-compliant.. if it is, we should change doc.
Thank you

**Comment by scanum** [comment]
@MikePlumleyMSFT Hi! Following up on this issue. Thanks!

**Comment by MikePlumleyMSFT** [comment]
@TroyStarr do you know?

**Comment by scanum** [comment]
@TroyStarr @MikePlumleyMSFT Following up on this issue. Thanks

**Comment by scanum** [comment]
@TroyStarr @MikePlumleyMSFT Following up on this issue. Thanks

**Comment by TroyStarr** [comment]
Hi @Mr-Nivek, I took a look and I believe the product is behaving correctly, and the documentation is also correct.  The "Claims to Windows Token Service" is compatible with the following MinRole server roles:

- Application
- Application with Search
- Custom
- Front-end
- Front-end with Distributed Cache
- Single-Server Farm

The "Claims to Windows Token Service" is not compatible with the following MinRole server roles:

- Distributed Cache
- Search

However, you'll need to be sure to [enable auto provisioning](https://docs.microsoft.com/sharepoint/administration/managing-a-minrole-server-farm-in-sharepoint-server-2016#manage-services-in-this-farm) of this service to indicate to MinRole that you want this service running in your farm.  I set up a SharePoint Server 2019 test farm with a server configured to use the "Single-Server Farm" server role.  I went to SharePoint Central Administration -> System Settings -> Manage services in this farm and clicked Enable Auto Provision on the Claims to Windows Token Service.  This enabled the Claims to Windows Token Service on the supported server roles.  I then ran the "Server role configuration isn't correct" health rule.  It succeeded with no compliance issues reported in the health analyzer report, and with the Claims to Windows Token Service still running on the supported server roles.

Note that we changed the behavior of the Claims to Windows Token Service starting in [SharePoint Server 2016 Feature Pack 1](https://docs.microsoft.com/sharepoint/what-s-new/new-features-in-feature-pack-1).  Before then, it would have been started by default and was compatible with all server roles.  After Feature Pack 1, it isn't started by default and we removed support for running it on the Distributed Cache and Search server roles.

You didn't mention what SharePoint version/build you're running, nor the steps you used to set up the Claims to Windows Token Service.  But if you're running SharePoint Server 2016 Feature Pack 1 or newer, and followed the steps I mentioned above to enable the service, and still have issues, I recommend opening a support case with Microsoft Support.  They can take a closer look at your environment to determine why you're seeing that behavior and work with you to fix it.
